v1.2/provider/provider-equinix-metal/report.log

```diff
@@ -0,0 +1,65 @@
+=== RUN   TestPackage
+=== RUN   TestPackage/IsInstalledAndHealthy
+    provider_test.go:69: Testing that the provider's Healthy and Installed status conditions are 'True'.
+    provider_test.go:85: Provider "equinix-provider-equinix-metal" is Healthy and Installed
+=== RUN   TestPackage/DeploysConformantCRDs
+    provider_test.go:93: Testing that ProviderRevision "equinix-provider-equinix-metal-5adc38d0eea7" delivers only conformant CRDs.
+=== RUN   TestPackage/DeploysConformantCRDs/ProviderConfig
+=== RUN   TestPackage/DeploysConformantCRDs/ProviderConfigUsage
+=== RUN   TestPackage/DeploysConformantCRDs/VirtualNetwork
+    provider_test.go:440: version "v1alpha1" OpenAPI schema: -want, +got:
+          &v1.JSONSchemaProps{
+          	... // 1 ignored and 19 identical fields
+          	MaxProperties: nil,
+          	MinProperties: nil,
+        - 	Required:      []string{"spec"},
```

Comment:
As far as I can tell, the spec definition and the parent types (Device, VirtualNetwork, Assignment) are defined in the same way as the GCP CloudSQLInstance type. I don't see extra kubebuilder 'required' tags in the GCP provider.

Any hints @negz?

Reply:
For comparison:

https://github.com/crossplane/provider-gcp/blob/master/apis/database/v1beta1/cloudsql_instance_types.go#L523-L552

https://github.com/crossplane-contrib/provider-equinix-metal/blob/master/apis/server/v1alpha2/device_types.go#L56-L89

Reply:
And the CRDs, where GCP includes `required: ["spec"]`:
https://github.com/crossplane/provider-gcp/blob/master/package/crds/database.gcp.crossplane.io_cloudsqlinstances.yaml#L422-L423

https://github.com/crossplane-contrib/provider-equinix-metal/blob/master/package/crds/server.metal.equinix.com_devices.yaml (does not include this)

Reply:
I don't see `kubebuilder:validation:Required` in the GCP provider, defining the required as the default for package-wide, but I do wonder if it is being set in another way.

Reply:
@displague I believe the `,omitempty` JSON tag is the culprit. iirc controller-tools falls back to that tag if there's no explicit required or optional comment markers. If the field has the `omitempty` JSON tag it assume it is optional.

Reply:
That will do it.
https://github.com/crossplane-contrib/provider-equinix-metal/pull/66